File: Data-Structures-And-Algorithms/BinarySearch/leetCode-69-mySqrt.go
func mySqrt(x int) int {
	l, r := 0, x
	ans := -1
	for l <= r {
		m := l + (r-l)/2
		if m*m <= x {
			ans = m
			l = m + 1
		} else {
			r = m - 1
		}
	}
	return ans
}

File: Data-Structures-And-Algorithms/BinarySearch/leetCode-69-mySqrt.java
class Solution {
    public int mySqrt(int x) {
        int l = 0, r = x, ret = -1;
        while (l <= r) {
            int m = l + (r-l)/2;
            if ((long)m * m <= x) {
                ret = m;
                l = m + 1;
            } else {
                r = m - 1;
            }
        }
        return ret;
    }
}
